### PR TITLE
python3Packages.pip-tools: fix build

### PR DESCRIPTION
**Pull request by misuzu**
###### Motivation for this change
https://hydra.nixos.org/build/157318353

ZHF: #144627

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by misuzu** [comment]
@GrahamcOfBorg build python3Packages.pip-tools

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 144837` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.pip-tools</li>
    <li>python39Packages.pip-tools</li>
  </ul>
</details>

File: pkgs/development/python-modules/pip-tools/default.nix [hunk 1]
```diff
@@ -24,14 +21,23 @@ buildPythonPackage rec {
...
 
-  checkPhase = ''
-    export HOME=$(mktemp -d) VIRTUAL_ENV=1
-    py.test -m "not network"
-  '';
+  propagatedBuildInputs = [
+    click
+    pep517
+    pip
+    setuptools-scm
```
Comment (fabaff) [inline pkgs/development/python-modules/pip-tools/default.nix:33]: Looks like that `setuptools-scm` could be [moved to `nativeBuildInputs`](https://github.com/jazzband/pip-tools/blob/master/setup.cfg#L29).

**Comment by risicle** [comment]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 144837 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>2 packages built:</summary>  <ul>    <li>python38Packages.pip-tools</li>    <li>python39Packages.pip-tools</li>  </ul></details>

**Pull request merged**